### PR TITLE
Improve doc for `jnp.logaddexp2`

### PR DESCRIPTION
**Pull request by rajasekharporeddy**
Part of #21461 

Rendered Preview:
- https://jax--23655.org.readthedocs.build/en/23655/_autosummary/jax.numpy.logaddexp2.html

### CODE REVIEW

File: jax/_src/numpy/ufuncs.py [hunk 1]
```diff
@@ -1744,11 +1744,39 @@ def _wrap_between(x, _a):
...
 
 
-@custom_jvp
-@implements(np.logaddexp2, module='numpy')
 @jit
 def logaddexp2(x1: ArrayLike, x2: ArrayLike, /) -> Array:
+  """Logarithm of the sum of exponentiations of the inputs in base-2.
+
+  JAX implementation of :obj:`numpy.logaddexp2`.
+
```
Comment (jakevdp) [inline jax/_src/numpy/ufuncs.py:1752]: We should mention somewhere that this is implemented carefully to work correctly in cases where the naive version would overflow.

Comment (rajasekharporeddy) [reply jax/_src/numpy/ufuncs.py:1752]: Changed the API description to `"Logarithm of the sum of exponentials of inputs in base-2 avoiding overflow."`

Comment (jakevdp) [reply jax/_src/numpy/ufuncs.py:1752]: Looks good, thanks!

**Comment by jakevdp** [comment]
Can you please squash the changes into a single commit? Thanks!

**Review by jakevdp — APPROVED** [review]
Thanks!

**Pull request merged**